httpx/dispatch/connection_pool.py

```diff
@@ -124,7 +126,13 @@ def num_connections(self) -> int:
             self.active_connections.remove(connection)
             self.max_connections.release()
             raise exc
-
+        except NoAvailableStreamIDError:
+            # NoAvailableStreamIDError propogates from http2.HTTP2Connection
+            # raised when stream ids expire
+            logger.debug("HTTP2 connection streams maxed out")
+            connection = await self.acquire_connection(origin=request.url.origin)
+            response = await connection.send(
```

Comment:
This line needs to be wrapped with the same "except BaseException as exc" behavior that we've got around the initial `connection.send()`. It's not obvious-at-a-glance to me what the easiest way of doing that is.

Reply:
@tomchristie Made the changes as requested. Could you please review if it looks okay? Also my test still doesn't work. I have no idea how to make request to the `MockHTTP2Server` I setup in my test. Thanks.